segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/AsyncBaseChunkStorage.java

```diff
@@ -313,11 +313,11 @@ public AsyncBaseChunkStorage(Executor executor) {
     final public CompletableFuture<Integer> read(ChunkHandle handle, long fromOffset, int length, byte[] buffer, int bufferOffset) {
         Exceptions.checkNotClosed(this.closed.get(), this);
         // Validate parameters
-        Preconditions.checkArgument(null != handle, "handle");
+        Preconditions.checkArgument(null != handle, "handle must not be null");
```

Comment:
Please change the null checks to one of the following:
1. `Preconditions.checkNotNull`
2. Add `@NonNull`  (from `Lombok`) as an annotation to the argument (preferred). This will save you from all this boilerplate code.

Reply:
NOTE - Preconditions.checkNull throws NullPointerException so there are places where InvalidArgmentException is need hence we do arg != null check. I have not changed those.

Reply:
Do not want to change method signature and tests for this change.